logstash-verifier/Dockerfile

```diff
@@ -1,55 +1,35 @@
-FROM msheiny/debian-jessie-systemd:latest
+FROM openjdk:8-jdk-slim-stretch
 
 MAINTAINER Freedom of the Press
-ENV GOPATH /tmp/logstash-filter-verifier
 
-ENV GOLANG_VERSION 1.8
-ENV GOLANG_DOWNLOAD_URL https://golang.org/dl/go$GOLANG_VERSION.linux-amd64.tar.gz
-ENV GOLANG_DOWNLOAD_SHA256 53ab94104ee3923e228a2cb2116e5e462ad3ebaeea06ff04463479d7f12d27ca
-ENV GOROOT /usr/local/go
-ENV PATH $GOROOT/bin:$PATH
+ENV LS_VERIFIER_TAR https://github.com/magnusbaeck/logstash-filter-verifier/releases/download/1.4.1/logstash-filter-verifier_1.4.1_linux_amd64.tar.gz
+ENV LS_VERIFIER_SHA256 2c322c84d5a82532ee7aee56f06bb1046b62fe46a73e920be970301382decaf1
+ARG LS_VER
+
 
 RUN apt-get update && apt-get install -y --no-install-recommends \
-        apt-transport-https \
-        binutils \
-        git \
-        make \
-        curl \
-        gnupg2 \
-        ca-certificates \
-        paxctl \
-    && rm -rf /var/lib/apt/lists/*
-
-RUN curl -fsSL "$GOLANG_DOWNLOAD_URL" -o golang.tar.gz && \
-    echo "$GOLANG_DOWNLOAD_SHA256  golang.tar.gz" | sha256sum -c - && \
-    tar -C /usr/local -xzf golang.tar.gz && \
-    rm golang.tar.gz && \
-    mkdir $GOPATH && \
-    cd $GOPATH && \
-    go get -d github.com/magnusbaeck/logstash-filter-verifier && \
-    cd src/github.com/magnusbaeck/logstash-filter-verifier && \
-    make install && \
-    cd /tmp && \
-    rm -rf $GOPATH && \
-    rm -rf $GOROOT
-
-RUN apt-get update && apt-get install paxctld -y
+        curl paxctl gnupg2 apt-transport-https && \
+    rm -rf /var/lib/apt/lists/* && \
+    rm -rf /var/cache/apt/*
 
 RUN curl -fs https://packages.elastic.co/GPG-KEY-elasticsearch | apt-key add - && \
-    echo "deb https://artifacts.elastic.co/packages/5.x/apt stable main" > /etc/apt/sources.list.d/elastic-5.x.list && \
-    echo "/usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java m" >> /etc/paxctld.conf && \
-    echo "/var/lib/dpkg/info/ca-certificates-java.postinst m" >> /etc/paxctld.conf && \
-    echo "/etc/ca-certificates/update.d/jks-keystore m" >> /etc/paxctld.conf && \
-    /etc/init.d/paxctld start && \
-    mkdir -p /usr/share/man/man1/ && \
-    apt-get update && \
-    apt-get install -y openjdk-8-jre-headless 2> /dev/null && \
-    apt-get install -y --allow-unauthenticated --no-install-recommends logstash && \
-    /usr/share/logstash/bin/logstash-plugin install logstash-filter-de_dot && \
-    rm -rf /var/lib/apt/lists/*
+    echo "deb https://artifacts.elastic.co/packages/6.x/apt stable main" > /etc/apt/sources.list.d/elastic.list && \
+    paxctl -cm /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java && \
+    apt-get update && apt-get install logstash=1:${LS_VER}-1 && \
+    rm -rf /var/lib/apt/lists/* && \
+    rm -rf /var/cache/apt/*
+
+RUN cd /tmp && \
+    curl -sL $LS_VERIFIER_TAR -o ls-verifier.tar.gz && \
+    echo "$LS_VERIFIER_SHA256 ls-verifier.tar.gz" | sha256sum -c - && \
+    tar -C /usr/bin -xzf ls-verifier.tar.gz
+
+RUN chown logstash /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java && \
```

Comment:
Wait, why does the `logstash` user need to _own_ the Java binary? It should be 755 by default.

Reply:
Good question! Becauussee i need to change the pax flags on the java binary prior to logstash execution. In order to do that, the `logstash` user needs write permissions on it.

You might be saying... well wait a second.. i thought you already changed the pax flags earlier in the dockerfile using root permissions!? You are correct -- but unfortunately however docker handles the filesystem doesnt retain these filesystem flags and they effectively get wiped upon a new docker run.

tl;dr - in order to run the container as non-root, we need pax flags set at docker runtime by the logstash user

Reply:
Thanks for clarifying, @msheiny, both here and in the appended comment. Let's dig more deeply on that separately, e.g. playing with `paxctl -c` versus `paxctl -C`.